### DISCUSSION TITLE
Add pinned hits conditionally

### DISCUSSION THREAD

**Discussion by skygatewitt** [algolia/instantsearch#5872] [Q&A] [answered]
Hi guys,

I want to display a certain set of pinned_hits in the infinite hits list, but only when no other filters or queries are active. Obviously, if I add them to the TypesenseInstantSearchAdapter in additionalSearchParameters, they are always present during search.

So I need a way to add them to the configuration conditionally after the fact. I can't seem to find a way. Neither middlewares, nor the deprecated searchFunction, nor an ais-configure tag did the trick so far.

Is there a way to do this without rules/overrides/curations beforehand?

Thanks in advance!

**Comment by Haroenv** [comment]
This is a question for TypeSense, not InstantSearch

**Reply by skygatewitt** [reply]
Not neccesarily, since inTypeSense I can do custom searches with different parameters each time. The "pinned_hits" ranking option in TypeSense is on the same level as "filter_by", which I am indeed able to change on the fly in Ais, or am I misinterpreting the parameter?

Edit:
My current solution is to create two search clients. One with the pinned hits, one without and with conditional rendering based on filter settings. It works, but it seems a bit hacky.

**Reply by Haroenv** [reply, marked as answer]
in Algolia you can do this with the builtin rules, which can have a condition on applied filters, so there's no need for this feature frontend

**Reply by skygatewitt** [reply]
Ok, I'll look into that. 

Thank you for your lightning fast answers! I really appreciate that :)
